src/hidimstat/desparsified_lasso.py

```diff
@@ -152,271 +122,378 @@ def desparsified_lasso(
     if dof_ajdustement:
-        coef_max = np.max(np.abs(beta_lasso))
-        support = np.sum(np.abs(beta_lasso) > 0.01 * coef_max)
+        coef_max = np.max(np.abs(beta_reid))
+        support = np.sum(np.abs(beta_reid) > 0.01 * coef_max)
         support = min(support, n_samples - 1)
         dof_factor = n_samples / (n_samples - support)
     else:
         dof_factor = 1
 
     # Computing Desparsified Lasso estimator and confidence intervals
-    beta_bias = dof_factor * np.dot(y.T, Z) / np.sum(X * Z, axis=0)
+    # Estimating the coefficient vector
+    beta_bias = dof_factor * np.dot(y_.T, Z) / np.sum(X_ * Z, axis=0)
 
-    P = ((Z.T.dot(X)).T / np.sum(X * Z, axis=0)).T
+    # beta hat
+    P = (np.dot(X_.T, Z) / np.sum(X_ * Z, axis=0)).T
     P_nodiag = P - np.diag(np.diag(P))
     Id = np.identity(n_features)
     P_nodiag = dof_factor * P_nodiag + (dof_factor - 1) * Id
-
-    beta_hat = beta_bias - P_nodiag.dot(beta_lasso)
-
+    beta_hat = beta_bias.T - P_nodiag.dot(beta_reid.T)
+    # confidence intervals
     omega_diag = omega_diag * dof_factor**2
-    omega_invsqrt_diag = omega_diag ** (-0.5)
 
+    if not group:
+        return beta_hat, sigma_hat, omega_diag
+    else:
+        cov_hat = sigma_hat
+        if cov is not None:
+            cov_hat = cov
+        theta_hat = n_samples * inv(cov_hat)
+        return beta_hat, theta_hat, omega_diag
+
+
+def desparsified_lasso_pvalue(
+    n_samples,
+    beta_hat,
+    sigma_hat,
+    omega_diag,
+    confidence=0.95,
+    distrib="norm",
+    eps=1e-14,
+    confidence_interval_only=False,
+):
+    """
+    Calculate confidence intervals and p-values for desparsified lasso estimators.
+    This function computes confidence intervals for the desparsified lasso
+    estimator beta_hat.
+    It can also return p-values derived from these confidence intervals.
+    Parameters
+    ----------
+    n_samples : float
+        The number of samples
+    beta_hat : array-like
+        The desparsified lasso coefficient estimates.
+    sigma_hat : float
+        Estimated noise level.
+    omega_diag : array-like
+        Diagonal elements of the precision matrix estimate.
+    confidence : float, optional (default=0.95)
+        Confidence level for intervals, must be in [0, 1].
+    distrib : str, optional (default="norm")
+        Distribution to use for p-value calculation.
+        Currently only "norm" supported.
+    eps : float, optional (default=1e-14)
+        Small value to avoid numerical issues in p-value calculation.
+    confidence_interval_only : bool, optional (default=False)
+        If True, return only confidence intervals.
+        If False, also return p-values.
+    Returns
+    -------
+    If confidence_interval_only=True:
+        cb_min : array-like
+            Lower bounds of confidence intervals
+        cb_max : array-like
+            Upper bounds of confidence intervals
+    If confidence_interval_only=False:
```

Comment:
Can we avoid this pattern of having different outputs. One solution is to always send the full version. Another possibility is to use a dictionary. But I think that systematically doing the full version is better?

Reply:
I added a boolean for choosing between the difference of outputs and this allows the user to choose what it wants. 
If you want to always have the full output, I can remove the boolean associate of these difference outputs.

Reply:
see issue #153

Reply:
done